src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java

```diff
@@ -323,33 +340,11 @@ private void processJibriIqFromMeet(JibriIq iq, XmppChatMember sender)
             if (jibriJid == null)
             {
                 sendErrorResponse(
-                    iq, XMPPError.Condition.service_unavailable, null);
+                        iq, XMPPError.Condition.service_unavailable, null);
                 return;
             }
 
-            final String roomName = MucUtil.extractName(senderMucJid);
-
-            try
-            {
-                final XMPPError err = startJibri(
-                        jibriJid, roomName, iq.getStreamId());
-                if (err == null)
-                {
-                    // ACK the original request
-                    sendResultResponse(iq);
-                }
-                else
-                {
-                    sendErrorResponse(
-                            iq, XMPPError.Condition.interna_server_error, null);
-                }
-            }
-            catch (final SmackException.NoResponseException e)
-            {
-                sendErrorResponse(
-                        iq, XMPPError.Condition.request_timeout, null);
-                return;
-            }
+            scheduleRetryStartJibri(jibriJid, iq);
```

Comment:
> There is no return after this statement and line 368 is always triggered which results in bad-request

:oops: fixed

> The whole else if should be changed to get rid of returns which are error prone(that is not related to the changes in the PR itself, but was like that from the start)

I've never really bought into the "one return at the end of the function" Java dogma in general, but as far as I know there's no way to ensure resources get cleaned up at the end of a particular scopes lifetime in Java, so maybe this is where the "returns are bad" thing comes from and it actually makes sense here? Or is there a way to insert a stack frame above the current one or something so that it will run when a function ends that I don't know about?